[libcap] build for arm failed

### Operating system

Linux

### Compiler

GCC

### Steps to reproduce the behavior

libcap being crosscompiled for arm unable to read file format Makefile

### Failure logs

[build-arm-linux-dynamic-dbg-err.log](https://github.com/user-attachments/files/17540977/build-arm-linux-dynamic-dbg-err.log)

[build-arm-linux-dynamic-dbg-out.log](https://github.com/user-attachments/files/17540976/build-arm-linux-dynamic-dbg-out.log)

If I set obcopy to arm-linux-gnueabihf-objcopy manually by editing Makefile
I got the following error: 
```
/bin/sh: 1: ./_makenames: Exec format error
make[1]: *** [Makefile:82: cap_names.h] Error 126
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile.vcpkg:13: all] Error 2
```

Because it was compiled for arm and can't be run on linux easily.

So, to resolve that it is also needed to set BUILD_CC to host triplet compiler
```
_makenames: _makenames.c cap_names.list.h
	$(BUILD_CC) $(BUILD_CFLAGS) $(BUILD_CPPFLAGS) $< -o $@
```
If I change this $(BUILD_CC) to gcc manually by editing Makefile, libcap configures and builds successfully

I assume the reason is [here](https://github.com/microsoft/vcpkg/blob/1b63f25c4cd58a699df0beb39474e62b2efc14a1/ports/libcap/configure#L22)

--- a/ports/libcap/configure
+++ b/ports/libcap/configure
@@ -21,6 +21,7 @@ BUILD_OPTIONS = \
     "AR=$AR" \
     "BUILD_CC=$CC" \
     "CC=$CC" \
+    "OBJCOPY=$OBJCOPY" \
     "RANLIB=$RANLIB" \
     "lib=lib" \
     "prefix=$prefix"

--- a/ports/libcap/portfile.cmake
+++ b/ports/libcap/portfile.cmake
@@ -2,7 +2,7 @@ vcpkg_download_distfile(ARCHIVE
     URLS "https://www.kernel.org/pub/linux/libs/security/linux-privs/libcap2/libcap-${VERSION}.tar.xz"
          "https://www.mirrorservice.org/sites/ftp.kernel.org/pub/linux/libs/security/linux-privs/libcap2/libcap-${VERSION}.tar.xz"
     FILENAME "libcap-${VERSION}.tar.xz"
-    SHA512 4e0bf0efeccb654c409afe9727b2b53c1d4da8190d7a0a9848fc52550ff3e13502add3eacde04a68a5b7bec09e91df487f64c5746ba987f873236a9e53b3d4e8
+    SHA512 59bb6781d96776595ad3df890f4e5188380634eabbb6128f3a5307946b01cf3bd19dee8a29d3e501de1d9e1c6ed0092c4cd5adc91da227a1260c1f4356cc0bf3
 )
 
 vcpkg_extract_source_archive(SOURCE_PATH

--- a/ports/libcap/portfile.cmake
+++ b/ports/libcap/portfile.cmake
@@ -18,6 +18,13 @@ if(VCPKG_CROSSCOMPILING)
     file(TOUCH "${SOURCE_PATH}/libcap/cap_names.h")
 endif()
 
+block(SCOPE_FOR VARIABLES)
+    set(VCPKG_CMAKE_CONFIGURE_OPTIONS "-DVCPKG_VARS_TO_CHECK=CMAKE_OBJCOPY")
+    vcpkg_cmake_get_vars(cmake_vars_file)
+    include("${cmake_vars_file}")
+    set(ENV{OBJCOPY} "${VCPKG_DETECTED_CMAKE_OBJCOPY}")
+endblock()
+
 vcpkg_configure_make(
     SOURCE_PATH "${SOURCE_PATH}"
     COPY_SOURCE

--- a/ports/libcap/vcpkg.json
+++ b/ports/libcap/vcpkg.json
@@ -9,6 +9,10 @@
     {
       "name": "libcap",
       "host": true
+    },
+    {
+      "name": "vcpkg-cmake-get-vars",
+      "host": true
     }
   ]
 }

--- a/ports/libcap/vcpkg.json
+++ b/ports/libcap/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "libcap",
-  "version": "2.70",
+  "version": "2.71",
   "description": "A library for getting and setting POSIX.1e (formerly POSIX 6) draft 15 capabilities.",
   "homepage": "https://sites.google.com/site/fullycapable/",
   "license": "BSD-3-Clause OR GPL-2.0-only",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4186,7 +4186,7 @@
     },
     "launch-darkly-server": {
       "baseline": "2.9.3",
-      "port-version": 0 
+      "port-version": 0
     },
     "lazy-importer": {
       "baseline": "2023-08-03",
@@ -4333,7 +4333,7 @@
       "port-version": 4
     },
     "libcap": {
-      "baseline": "2.70",
+      "baseline": "2.71",
       "port-version": 0
     },
     "libcbor": {

--- a/versions/l-/libcap.json
+++ b/versions/l-/libcap.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "624965e04fa91928fc26f8a061527d30ed58bf98",
+      "version": "2.71",
+      "port-version": 0
+    },
+    {
       "git-tree": "c7a0cc03982a375c59205880aea11305d37464c0",
       "version": "2.70",
       "port-version": 0

